client/src/pages/donation/settings.js

```diff
@@ -0,0 +1,209 @@
+import React, { Component, Fragment } from 'react';
+import Helmet from 'react-helmet';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { Grid, Row, Col, Button } from '@freecodecamp/react-bootstrap';
+import { uniq } from 'lodash';
+
+import { apiLocation } from '../../../config/env.json';
+import { postCreateHmacHash } from '../../utils/ajax';
+import {
+  signInLoadingSelector,
+  userSelector,
+  hardGoTo as navigate,
+  isSignedInSelector
+} from '../../redux';
+// eslint-disable-next-line max-len
+import DonateServicebotEmbed from '../../components/Donation/components/DonateServicebotEmbed';
+import { Loader, Spacer, Link } from '../../components/helpers';
+
+const propTypes = {
+  donationEmails: PropTypes.array,
+  email: PropTypes.string,
+  isDonating: PropTypes.bool,
+  isSignedIn: PropTypes.bool,
+  navigate: PropTypes.func.isRequired,
+  showLoading: PropTypes.bool.isRequired
+};
+
+const mapStateToProps = createSelector(
+  isSignedInSelector,
+  userSelector,
+  signInLoadingSelector,
+  (isSignedIn, { email, isDonating, donationEmails }, showLoading) => ({
+    isSignedIn,
+    email,
+    isDonating,
+    donationEmails,
+    showLoading
+  })
+);
+
+const mapDispatchToProps = {
+  navigate
+};
+
+export class DonationSettingsPage extends Component {
+  constructor(...props) {
+    super(...props);
+
+    this.state = {
+      hash: null,
+      currentSettingsEmail: null
+    };
+
+    this.getEmailHmacHash = this.getEmailHmacHash.bind(this);
+    this.handleSelectDonationEmail = this.handleSelectDonationEmail.bind(this);
+  }
+
+  getEmailHmacHash(currentSettingsEmail) {
+    return postCreateHmacHash({
+      email: currentSettingsEmail
+    })
+      .then(response => {
+        const data = response && response.data;
+        this.setState({ hash: '' + data.hash, currentSettingsEmail });
+      })
+      .catch(error => {
+        const data =
+          error.response && error.response.data
+            ? error.response.data
+            : {
+                error:
+                  'Something is not right. Please contact team@freecodecamp.org'
+              };
+        console.error(data.error);
+      });
+  }
+
+  handleSelectDonationEmail(e) {
+    e.preventDefault();
+    this.setState({ hash: null, currentSettingsEmail: null });
+    this.getEmailHmacHash(e.currentTarget.value);
+  }
+
+  renderServicebotEmbed() {
+    const { currentSettingsEmail, hash } = this.state;
+    if (!hash && !currentSettingsEmail) {
+      return null;
+    }
+    return (
+      <div>
+        <Spacer />
+        <DonateServicebotEmbed email={currentSettingsEmail} hash={hash} />
+      </div>
+    );
+  }
+
+  renderDonationEmailsList() {
+    const { donationEmails } = this.props;
+    return (
+      <div>
+        {uniq(donationEmails).map((email, index) => (
```

Comment:
```suggestion
        {uniq(donationEmails).map(email => (
```
Emails are already unique keys